src/transformers/models/conditional_detr/modeling_conditional_detr.py

```diff
@@ -378,7 +378,12 @@ def __init__(self, config):
             self.model.feature_info.channels() if config.use_timm_backbone else self.model.channels
         )
 
-        backbone_model_type = config.backbone if config.use_timm_backbone else config.backbone_config.model_type
+        backbone_model_type = None
+        if config.backbone is not None:
+            backbone_model_type = config.backbone
+        elif config.backbone_config is not None:
+            backbone_model_type = config.backbone_config.model_type
```

Comment:
> Is there a chance that backbone and backbone_config are both None by misconfiguration? 

It shouldn't happen from a fresh config, as `verify_backbone_config_arguments` will check that. 

It's possible something like this could happen: 
```py
from transformers import ConditionalDetrConfig, ConditionalDetrModel

config = ConditionalDetrConfig()
config.backbone = None
config.backbone_config = None

model = ConditionalDetrModel(config)
```
as it's not uncommon to modify configs post-creation. 

> If backbone_model_type will remain None, then if "resnet" in backbone_model_type: will raise an error:

Good point. I'll add in an exception is neither are set here and for the other, similar bits of logic